test/onnx/test_diagnostics.py

```diff
@@ -0,0 +1,287 @@
+# Owner(s): ["module: onnx"]
+
+import contextlib
+import dataclasses
+import io
+from typing import AbstractSet, Tuple
+
+import torch
+from torch.onnx import errors
+from torch.onnx._internal import diagnostics
+from torch.onnx._internal.diagnostics import infra
+from torch.testing._internal import common_utils
+
+
+def _assert_has_diagnostics(
+    engine: infra.DiagnosticEngine,
+    rule_level_pairs: AbstractSet[Tuple[infra.Rule, infra.Level]],
+):
+    sarif_log = engine.sarif_log()
+    unseen_pairs = {(rule.id, level.value) for rule, level in rule_level_pairs}
+    actual_results = []
+    for run in sarif_log.runs:
+        for result in run.results:
+            id_level_pair = (result.rule_id, result.level)
+            unseen_pairs.discard(id_level_pair)
+            actual_results.append(id_level_pair)
+
+    if unseen_pairs:
+        raise AssertionError(
+            f"Expected diagnostic results of rule id and level pair {unseen_pairs} not found. "
+            f"Actual diagnostic results: {actual_results}"
+        )
+
+
+@contextlib.contextmanager
+def assertAllDiagnostics(
```

Comment:
I recall following the unittest convention, thoughts?

Reply:
I don't think we need to follow its convention - unit test was that way presumably because it was inspired from a Java library. New method we create can follow pep just fine